File: src/com/liubs/aicommit/ai/AiClient.java
package com.liubs.aicommit.ai;

import com.intellij.openapi.progress.ProgressIndicator;
import com.liubs.aicommit.settings.ProviderProfile;
import org.jetbrains.annotations.Nullable;

import java.io.IOException;
import java.util.List;

public interface AiClient {

    /** 根据 diff 生成提交信息 */
    String generateCommitMessage(ProviderProfile profile, String apiKey, String diff,
                                 @Nullable ProgressIndicator indicator) throws IOException;

    /** 发起最小请求验证当前模型可用。 */
    String ping(ProviderProfile profile, String apiKey,
                @Nullable ProgressIndicator indicator) throws IOException;

    /** 从上游拉取可用模型列表(GET {baseUrl}/models) */
    List<String> listModels(String baseUrl, String apiKey,
                            @Nullable ProgressIndicator indicator) throws IOException;
}

File: vscode-plugin/src/ai/aiClient.ts
import * as vscode from 'vscode';
import { ProviderProfile } from '../settings/providerProfile';

export interface AiClient {
  generateCommitMessage(
    profile: ProviderProfile,
    apiKey: string,
    diff: string,
    token?: vscode.CancellationToken,
  ): Promise<string>;

  /** 测试连接用:发一个最小请求,返回模型的简短回复 */
  ping(
    profile: ProviderProfile,
    apiKey: string,
    token?: vscode.CancellationToken,
  ): Promise<string>;

  listModels(
    baseUrl: string,
    apiKey: string,
    token?: vscode.CancellationToken,
  ): Promise<string[]>;
}
